### PR TITLE
Made it possible to define custom converters

### PR DESCRIPTION
**Pull request by dlmr**
Also made the code more modular

### CODE REVIEW

**Comment by bjarneo** [comment]
groupt1 = group1?

**Comment by dlmr** [comment]
@bjarneo Woops, thanks!

File: docs/config/settings.md [hunk 1]
```diff
@@ -85,3 +93,10 @@ Roc assumes that the validators used is either a RegExp or a function that will
 For convenience several types of validators exists in `roc` that can be imported from `roc/validators`. For a complete list of them please see [the JSDocs](/docs/JSDocs.md).
 
 See [roc.config.meta.js in roc-web for an example of how the configuration might look](https://github.com/vgno/roc-web/blob/master/src/roc/config/roc.config.meta.js#L62).
+
+### `converters`
+Roc will by default provide an automatic converter that will base its converter on the default value of the settings. In most cases will this be sufficient but in some instances a custom converter might be needed. This custom converter will in most cases get a string as input and it's expected to return the correct type.
```
Comment (andreasrs) [inline docs/config/settings.md]: Roc will by default provide an automatic converter that will be based on the default value of the corresponding settings. In most cases will this be sufficient but in some instances a custom converter might be needed. This custom converter will in most cases get a string as input and be expected to return the correct type.

File: src/documentation/helpers.js [hunk 2]
```diff
@@ -35,10 +35,11 @@ export function addPadding(string, length) {
  */
 export function toCliOption(configPaths) {
     // Runtime should be added directly
-    if (configPaths[0] === 'runtime') {
-        configPaths.shift();
-    }
-    return '--' + configPaths.join('-');
+    const paths = configPaths[0] === 'runtime' ?
+        configPaths.slice(1, configPaths.length) :
```
Comment (andreasrs) [inline src/documentation/helpers.js]: Don't think argument 2 of slice is needed here

**Comment by andreasrs** [comment]
Only minor things. LGTM otherwise :+1: 
Seems to work nicely.

File: src/documentation/helpers.js (hunk 2 above)
Comment (dlmr) [inline src/documentation/helpers.js]: Correct, I will fix this!

**Pull request merged**